File: TerminBA/TerminBA.Models/Request/PostInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TerminBA.Models.Request
{
    public class PostInsertRequest
    {
        [Required]
        public string? SkillLevel { get; set; }

        [MaxLength(100)]
        public string? Text { get; set; }

        [Required]
        public int ReservationId { get; set; }

        [Required]
        public int NumberOfPlayersWanted { get; set; }

    }
}

File: TerminBA/TerminBA.UI/terminba_mobile/lib/model/post_insert_request.dart
class PostInsertRequest {
  final int skillLevelId;
  final String? text;
  final int reservationId;
  final int numberOfPlayersWanted;

  const PostInsertRequest({
    required this.skillLevelId,
    this.text,
    required this.reservationId,
    required this.numberOfPlayersWanted,
  });

  Map<String, dynamic> toJson() => {
        'skillLevelId': skillLevelId,
        'text': text,
        'reservationId': reservationId,
        'numberOfPlayersWanted': numberOfPlayersWanted,
      };
}
